pkg/traceql/ast.go

```diff
@@ -305,7 +304,7 @@ func newSpansetFilter(e FieldExpression) *SpansetFilter {
 func (*SpansetFilter) __spansetExpression() {}
 
 func (f *SpansetFilter) evaluate(input []*Spanset) ([]*Spanset, error) {
-	f.outputBuffer = f.outputBuffer[:0]
+	var outputBuffer []*Spanset
```

Comment:
This change is required to make the traceql eval callback safe for concurrent use (so we can execute the query against multiple blocks in parallel).  This was ok previously because: regular searches were single threaded or the query was compiled for each block and they got their own AST (and SpansetFilters). But now for metrics the AST keeps more state and it should be more efficient to reuse the same AST across all blocks.  Otherwise it would have to create and squash repeatedly the same timeseries from each block to get the final results.